File: Ruby/lib/ltpservice.rb
# ltpservice.rb

$addr = 'api.ltp-cloud.com:8080'
$port = '80'
$uris = '/ltp_srv/ltp'

require 'net/http'
require 'uri'

class LTPService
    def initialize(username, password)
        @username = username
        @password = password
    end

    def analyze(input, opt = nil, encoding = 'UTF-8')
        url = URI('http://' + $addr + ':' + $port + $uris)
        req = Net::HTTP::Post.new(url.path)
        req.basic_auth @username, @password

        if input.kind_of? String
            req.set_form_data({'s' => input,
                            'c' => encoding, 
                            't' => (opt.nil? ? LTPOption::ALL : opt)})

            res = Net::HTTP.new(url.host, url.port).start do |http|
                http.request(req)
            end
        elsif input.kind_of? LTML
            #puts input.to_s
            req.set_form_data({'s' => input.to_s,
                            'x' => 'y',
                            'c' => encoding, 
                            't' => (opt.nil? ? LTPOption::ALL : opt)})

            res = Net::HTTP.new(url.host, url.port).start do |http|
                http.request(req)
            end
        end

        LTML.new(res.body)
    end
end

require_relative 'ltpservice/LTML'
require_relative 'ltpservice/LTPOption'

File: Python/ltpservice/ltpservice.py
#!/usr/bin/env python
import urllib, urllib2, base64
import LTPOption

from LTML import LTML

addr = "http://api.ltp-cloud.com:8080"
uris = "/ltp_srv/ltp"

class LTPService(object):
    """
    ltp service class
    """
    #   @param[in]  au      authorization token
    def __init__(self, au, encoding="utf-8"):
        self.au = au
        self.encoding = encoding;

    #
    # execute the analysis process
    #
    #   @param[in]  input   the input sentence
    #   @param[in]  opt     the options
    #   @return     LTML    the result
    def analysis(self, input, opt = None):

        request = urllib2.Request(addr + uris);
        request.add_header("Authorization", 
                "Basic %s" % (base64.encodestring(self.au).replace("\n", "")))

        if isinstance(input, str):
            data = {
                    's': input,
                    'x': 'n',
                    'c': self.encoding,
                    't': (opt if opt is not None else LTPOption.ALL)}

            params = urllib.urlencode(data)
            result = urllib2.urlopen(request, params)
            content = result.read().strip()
            return LTML(content)

        elif isinstance(input, LTML):
            data = {
                    's' : input.tostring(self.encoding),
                    'x' : 'y',
                    'c' : self.encoding,
                    't' : (opt if opt is not None else LTPOption.ALL),}

            params = urllib.urlencode(data)
            result = urllib2.urlopen(request, params)
            content = result.read().strip()
            return LTML(content)
